homeassistant/components/switch/vesync.py

```diff
@@ -0,0 +1,110 @@
+"""
+Support for Etekcity VeSync switches.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.vesync/
+"""
+import logging
+import voluptuous as vol
+from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (CONF_USERNAME, CONF_PASSWORD)
+import homeassistant.helpers.config_validation as cv
+
+
+REQUIREMENTS = ['pyvesync==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+API_BASE_URL = 'https://smartapi.vesync.com'
+API_RATE_LIMIT = 30
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the VeSync switch platform."""
+    from pyvesync.vesync import VeSync
+
+    switches = []
+
+    try:
+        manager = VeSync(config.get(CONF_USERNAME), config.get(CONF_PASSWORD))
+    except ValueError:
+        _LOGGER.error("Unable to setup VeSync platform")
+    else:
```

Comment:
You can save on indentation if you `return` inside the `except` and axe the `else`

Reply:
Agreed. Cleaned up.